src/components/include/security_manager/security_manager.h

```diff
@@ -120,18 +125,34 @@ class SecurityManager : public protocol_handler::ProtocolObserver {
    * identifier
    * @return new \c  SSLContext or \c NULL on any error
    */
-  virtual SSLContext* CreateSSLContext(const uint32_t& connection_key) = 0;
+  virtual SSLContext* CreateSSLContext(const uint32_t& connection_key,
+                                       ContextCreationStrategy cc_strategy) = 0;
 
   /**
    * \brief Start handshake as SSL client
    */
   virtual void StartHandshake(uint32_t connection_key) = 0;
 
+  /**
+   * @brief PostponeHandshake allows to postpone handshake. It notifies
+   * cryptomanager that certificate should be updated and adds specified
+   * connection key to the list of the certificate awaiting connections.
+   * @param connection_key the identifier for connection to postpone handshake.
+   */
+  virtual void PostponeHandshake(const uint32_t connection_key) = 0;
+
   /**
    * @brief Check whether certificate should be updated
    * @return true if certificate should be updated otherwise false
    */
-  virtual bool IsCertificateUpdateRequired() = 0;
+  virtual bool IsCertificateUpdateRequired(const uint32_t connection_key) = 0;
```

Comment:
@AByzhynar please update description.

Reply:
@LuxoftAKutsan Changed in eb15dd852a0ea418bfa444b74c8f15d792e254ad